lms/djangoapps/student_account/views.py

```diff
@@ -587,3 +591,38 @@ def account_settings_context(request):
         } for state in auth_states if state.provider.display_for_login or state.has_account]
 
     return context
+
+
+class RecoverPasswordView(ViewSet):
+    """Resets a password of a user by proving the email address
+
+        Example Requests:
+            POST /account/recover-password
+            {"email": "staff@example.com"}
+
+        Response Values:
+            HttpResponse: 200 if the password was reset correctly.
+            HttpResponse: 400 if email wasn't provided.
+            HttpResponse: 404 if user was not found with provided email.
+    """
+
+    def post(self, request, format=None):
+        email = request.data.get('email')
+        if email:
+            limiter = BadRequestRateLimiter()
+            if limiter.is_rate_limit_exceeded(request):
+                AUDIT_LOG.warning("Password reset rate limit exceeded")
+                return HttpResponseForbidden()
+
+            try:
+                request_password_change(email, request.get_host(), request.is_secure())
+                user = User.objects.get(email=email)
```

Comment:
Can we move this line to the start of the try-catch block? That way we can catch a non-existing user for the email faster, rather than having to let `request_password_change` do it.